### PR TITLE
monitoring: grafana email notifier support

### PR DESCRIPTION
**Pull request by bobheadxi**
Closes https://github.com/sourcegraph/sourcegraph/issues/11454

This PR adds:

* email support for notifiers
* automatic sync for SMTP settings from Sourcegraph site configuration

Implementation details:

* map relevant parts of all site config to a `GrafanaChange`
* changes now only have access to the new configuration (ie their goal is purely to apply incoming changes - this ties into my first question further down)
* add new wrapper around the grafana client, `GrafanaController`, to also allow stop/restart capabilities

Questions:

* [x] failure behaviour (https://github.com/sourcegraph/sourcegraph/issues/11454#issuecomment-644576628 => https://github.com/sourcegraph/sourcegraph/pull/11554/files#diff-42ca82e5e966d29f64bb4a21c9e25983R194), currently:
    * nothing happens if problem is encountered by `grafana-wrapper` while performing update
    * if grafana exits with status code >2, `grafana-wrapper` exits (exit code 1 is emitted when Grafana is SIGINT'd so we ignore it)
* [x] is EHLO == HELO? https://github.com/sourcegraph/sourcegraph/pull/11554/files#diff-229cb10705aabb2149674e4fd1a58bfeR30
* [x] not sure what `smtp.authentication` does (don't seem to see an equivalent in the [Grafana SMTP config](https://grafana.com/docs/grafana/latest/installation/configuration/#smtp)) => I believe Grafana tries `CRAM-MD5 PLAIN` in that order

### CODE REVIEW

**Comment by unknwon** [comment]
Is this ready for review? Seems still have many TODOs left. If not ready yet, suggest convert to draft :)

image

**Comment by bobheadxi** [comment]
Ah good idea @unknwon sorry about that!

File: docker-images/grafana/cmd/grafana-wrapper/smtp.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+func newGrafanaSMTPConfig(email *siteEmailConfig) *grafanaConfigSMTP {
+	if email == nil || email.SMTP == nil {
+		return &grafanaConfigSMTP{Enabled: false}
+	}
+	return &grafanaConfigSMTP{
+		Enabled:      true,
+		Host:         fmt.Sprintf("%s:%d", email.SMTP.Host, email.SMTP.Port), // host, port are required
+		User:         email.SMTP.Username,
+		Password:     email.SMTP.Password,
+		EhloIdentity: email.SMTP.Domain, // TODO: is this right?
```
Comment (slimsag) [inline docker-images/grafana/cmd/grafana-wrapper/smtp.go:30]: this is right.

**Comment by bobheadxi** [comment]
Converting the draft for now due to change in plans: https://github.com/sourcegraph/sourcegraph/issues/11452#issuecomment-648652514 (will close when I have a new PR up)

**Pull request closed without merging**